### PR TITLE
Add a lock file during the install (and it's removed only if the install is well finished)

### LINKED ISSUE #28499: When accessing admin during install, we cannot know if install is well finished
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

During the conception of the edition, we find a problem.

* You make the install. 
* During the install (progress bar), you can access to admin directory but you cannot know (from hook modules) if the install is well finished.

### Describe the solution you'd like

A configuration value where we define if the installation is well finished.

### Alternatives you've considered

_No response_

### Additional context

_No response_

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add a lock file during the install
| Type?             | improvement
| Category?         | IN
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28499
| How to test?      | * Install (normally) and the lock file `var/.install.prestashop` is removed<br />* Install (with making an error) and the lock file `var/.install.prestashop` remains present

### CODE REVIEW

**Review by matks — CHANGES_REQUESTED** [review]
I mentioned some other solution https://github.com/PrestaShop/PrestaShop/issues/28499#issuecomment-1128122567

**Review by atomiix — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by AureRita** [comment]
Hi @Progi1984 

Thank you for your PR, I tested it and it've a little problem, the file isn't create during a "normal" installation, it will be created when you do a "restart" installation, so at the end of a "normal" installation, that will create red notice to pass in a "restart" installation, I think it's not what it supposed to do, you can see this explanation on this video : 

https://user-images.githubusercontent.com/100699445/179959703-6ebdc495-bc50-4f28-b1f3-9b63d895ec25.mp4

Thank you

**Comment by Progi1984** [comment]
@AureRita I fixed the bug and the "How to test?"

@PrestaShop/prestashop-maintainers & @PrestaShop/committers Hi, could you remove the label "Waiting for author", please ? And if you have time for review :heart_eyes:

**Review by marsaldev — APPROVED** [review]

File: controllers/admin/AdminLoginController.php
Comment (zuk3975) [inline controllers/admin/AdminLoginController.php:107]: Just want to make sure I get it all right (I don't yet fully understand usecase/purpose of this).

so if install failed,
the lock file will still exist and then
the `admin` dir won't be renamed.
So what happens next? it redirects somewhere, but it doesn't show any errors or anything like that about the unfinished install right? (so from UX point there are no changes).

So we just leave the lock file for module hooks (or basically for developers) to check its existence when needs to determine if install finished? (but when is it needed?, could you provide a brief example usecase?, I didn't get it from PR description fully)

Comment (Progi1984) [reply controllers/admin/AdminLoginController.php:107]: @zuk3975 Have you checked the linked issue ?

Comment (zuk3975) [reply controllers/admin/AdminLoginController.php:107]: yes, sorry, but I didn't really get it fully when/who accesses the shop during the installation (therefore I wanted to get the example of a usecase, so Im sure we are on the same page).

Comment (Progi1984) [reply controllers/admin/AdminLoginController.php:107]: **Who ?** The Shop Administrator
**When ?** During the install
**What ?** The SA wants to access to the directory `admin` during the install

File: src/PrestaShopBundle/Install/Install.php
Comment (zuk3975) [inline src/PrestaShopBundle/Install/Install.php:1021]: just a note - Instead of such comments which IMO doesn't add any benefit (because the code is verbose enough and talks for itself), I would rather appreciate the explanation of what is that lock file purpose above the constant itself

File: controllers/admin/AdminLoginController.php
Comment (zuk3975) [reply controllers/admin/AdminLoginController.php:107]: okay, sorry, but then shouldn't it be handled somewhere - like some error shown that "installation is still in progress" in that case? Because I get fatal errors if I try to reach not fully installed prestashop (tables/configs missing etc.), so isn't that part still missing in PR? (Im confused that - the PR basically avoids renaming admin dir "because merchant may wanna reach it during installation", but merchant will still get a fatal error when reaching `/admin` with this PR or without it. So from the issue and pr description I can't understand the overall purpose of this. Is it just `introducing a way to determine if install is complete` or is it also `to avoid some errors when reaching "/admin" during installation` (because this part is still missing then)?

Comment (Progi1984) [reply controllers/admin/AdminLoginController.php:107]: It's the part to determine if install is complete. That avoid some errors when reaching "/admin" during installation (but not all).

Comment (zuk3975) [reply controllers/admin/AdminLoginController.php:107]: okay then

**Review by zuk3975 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @Progi1984 

Thank you for your PR, I tested it and it works, as we can see here : 

[Untitled_ Oct 26, 2022 4_09 PM.webm](https://user-images.githubusercontent.com/100699445/198291027-d84cd5b9-b838-43c2-8239-145ab8524dc7.webm)

even in command line : 

[Untitled_ Oct 27, 2022 3_03 PM.webm](https://user-images.githubusercontent.com/100699445/198292074-7d834a40-6321-4a29-b52c-4e579370521b.webm)

**Pull request merged**
